sql/src/main/java/org/apache/druid/sql/avatica/ErrorHandler.java

```diff
@@ -0,0 +1,118 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.sql.avatica;
+
+import com.google.inject.Inject;
+import org.apache.druid.common.exception.ErrorResponseTransformStrategy;
+import org.apache.druid.common.exception.NoErrorResponseTransformStrategy;
+import org.apache.druid.common.exception.SanitizableException;
+import org.apache.druid.java.util.common.ISE;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.query.QueryException;
+import org.apache.druid.query.QueryInterruptedException;
+import org.apache.druid.server.initialization.ServerConfig;
+import org.apache.druid.server.security.ForbiddenException;
+
+import javax.annotation.Nonnull;
+
+/**
+ * ErrorHandler is a utilty class that is used to sanitize and log exceptions.
+ */
+class ErrorHandler
+{
+  private static final Logger LOG = new Logger(DruidMeta.class);
+  private final ErrorResponseTransformStrategy errorResponseTransformStrategy;
+
+  @Inject
+  ErrorHandler(final ServerConfig serverConfig)
+  {
+    this.errorResponseTransformStrategy = serverConfig.getErrorResponseTransformStrategy();
+  }
+
+  /**
+   * Logs a throwable at the error level and sanitizes the throwable if applicable. Will return
+   * the sanitized or original throwable.
+   *
+   * @param error A throwable that will be logged then sanitized
+   * @param <T>
+   * @return The sanitized throwable
+   */
+  public <T extends Throwable> RuntimeException logFailureAndSanitize(T error)
+  {
+    return logFailureAndSanitize(error, error.getMessage());
+  }
+
+  /**
+   * Logs an error message at the error level and sanitizes the throwable if applicable. Will return
+   * the sanitized or original throwable.
+   *
+   * @param error   the throwable that will be sanitized
+   * @param message the error string formate message to be logged
+   * @param format  the format args for the message
+   * @param <T>
+   * @return A sanitized version of the throwable if applicable otherwise the original throwable
+   */
+  public <T extends Throwable> RuntimeException logFailureAndSanitize(T error, String message, Object... format)
+  {
+    logFailure(error, message, format);
+    return sanitize(error);
+  }
+
+  public <T extends Throwable> T logFailure(T error, String message, Object... format) {
+    LOG.error(error, message, format);
+    return error;
+  }
+
+  public <T extends Throwable> T logFailure(T error) {
+    logFailure(error, error.getMessage());
+    return error;
+  }
+
+  public <T extends Throwable> RuntimeException sanitize(T error)
+  {
+    if (error instanceof QueryInterruptedException) {
```

Comment:
Do we need to check for UOE too? Also should UOE implement SanitizableException?

Reply:
No caller of this throwing a UOE explicitly, but it doesn't hurt to add it imo